frontend/src/components/beta/accounts/AccountActivityView.tsx

```diff
@@ -0,0 +1,485 @@
+  return `/beta/accounts/${account}`;
+};
+
+const ActivityItemActionWrapper = styled("div", {
+  display: "flex",
+  alignItems: "center",
+  whiteSpace: "pre",
+
+  "& + &": {
+    marginTop: 4,
+  },
+});
+
+const ActivityItemTitle = styled("span", {
+  fontWeight: "bold",
+});
+
+const ActivityItemAction: React.FC<{
+  action: AccountActivityRelatedAction | AccountActivityAction;
+}> = ({ action }) => {
+  const badge = (
+    <>
+      {"sender" in action ? (
+        <>
+          <Link href={getAccountLink(action.sender)}>
+            <a>{shortenString(action.sender)}</a>
+          </Link>
+          {" → "}
+        </>
+      ) : null}
+      <AccountActivityBadge
+        action={action}
+        href={"transactionHash" in action ? getActionLink(action) : undefined}
+      />
+      {"receiver" in action ? (
+        <>
+          {" → "}
+          <Link href={getAccountLink(action.receiver)}>
+            <a>{shortenString(action.receiver)}</a>
+          </Link>
+        </>
+      ) : null}
+    </>
+  );
+  switch (action.type) {
+    case "transfer": {
+      const deltaAmount = JSBI.BigInt(action.deltaAmount);
+      return (
+        <ActivityItemActionWrapper>
+          {badge}{" "}
+          {JSBI.equal(deltaAmount, BI.zero) ? null : (
+            <>
+              {" 💸 "}
+              <NearAmount amount={action.deltaAmount} />
+            </>
+          )}
+        </ActivityItemActionWrapper>
+      );
+    }
+    case "restake": {
+      const deltaAmount = JSBI.BigInt(action.deltaAmount);
+      return (
+        <ActivityItemActionWrapper>
+          {badge}{" "}
+          {JSBI.equal(deltaAmount, BI.zero) ? null : (
+            <>
+              {" 💸 "}
+              <NearAmount amount={action.deltaAmount} />
+            </>
+          )}
+        </ActivityItemActionWrapper>
+      );
+    }
+    case "call-method": {
+      const attachedAmount = JSBI.BigInt(action.attachedAmount);
+      return (
+        <ActivityItemActionWrapper>
+          {badge}{" "}
+          {JSBI.equal(attachedAmount, BI.zero) ? null : (
+            <>
+              {" 💸 "}
+              <NearAmount amount={action.attachedAmount} />
+            </>
+          )}
+        </ActivityItemActionWrapper>
+      );
+    }
+    default:
+      return <ActivityItemActionWrapper>{badge}</ActivityItemActionWrapper>;
+  }
+};
+
+const ActivityItemRow: React.FC<RowProps> = ({ item, options }) => {
+  const { t } = useTranslation();
+  const deltaAmount = JSBI.BigInt(item.deltaAmount);
+  const isDeltaAmountZero = JSBI.equal(deltaAmount, BI.zero);
+  const isDeltaAmountPositive = JSBI.greaterThan(deltaAmount, BI.zero);
+  const absoluteDeltaAmount = isDeltaAmountPositive
+    ? deltaAmount
+    : JSBI.multiply(deltaAmount, BI.minusOne);
+  const actions =
+    item.action.type === "batch" ? item.action.actions : [item.action];
+
+  return (
+    <>
+      {actions.map((subaction, subindex) => {
```

Comment:
How about
```suggestion
      {actions.map((subAction, subIndex) => {
```
Way more readable I guess.